nomenclature/iamc/datapoint.py

```diff
@@ -0,0 +1,37 @@
+from typing import List
+from pydantic import BaseModel, field_validator
+
+
+from nomenclature.definition import DataStructureDefinition
+
+IAMC_DIMENSIONS = ["model", "scenario", "region", "variable", "unit"]
+
+
+class IamcDataFilter(BaseModel):
+    model: List[str] | None = None
+    scenario: List[str] | None = None
+    region: List[str] | None = None
+    variable: List[str] | None = None
+    unit: List[str] | None = None
+    year: List[int] | None = None
+
+    @field_validator(*IAMC_DIMENSIONS, mode="before")
```

Comment:
I think you're not applying the validator to `year` since it's not in `IAMC_DIMENSIONS`.